src/main/java/edu/cornell/kfs/coa/batch/dataaccess/impl/WorkdayOpenAccountDaoJdbc.java

```diff
@@ -0,0 +1,167 @@
+    protected String ACCOUNT_NBR = "ACCOUNT_NBR";
+    protected String ACCOUNT_NM = "ACCOUNT_NM";
+    protected String SUB_FUND_GRP_WAGE_IND = "SUB_FUND_GRP_WAGE_IND";
+    protected String SUB_FUND_GRP_CD = "SUB_FUND_GRP_CD";
+    protected String FIN_HGH_ED_FUNC_CD = "FIN_HGH_ED_FUNC_CD";
+    protected String ACCT_EFFECT_DT = "ACCT_EFFECT_DT";
+    protected String ACCT_CLOSED_IND = "ACCT_CLOSED_IND";
+    protected String ACCT_TYP_CD = "ACCT_TYP_CD";
+    protected String SUB_ACCT_NBR = "SUB_ACCT_NBR";
+    protected String SUB_ACCT_NM = "SUB_ACCT_NM";
+    protected String SUB_ACCT_ACTV_CD = "SUB_ACCT_ACTV_CD";
+    protected String FIN_OBJECT_CD = "FIN_OBJECT_CD";
+    protected String FIN_SUB_OBJ_CD = "FIN_SUB_OBJ_CD";
+    protected String FIN_SUB_OBJ_CD_NM = "FIN_SUB_OBJ_CD_NM";
+    
+    
+    protected UniversityDateService universityDateService;
+
+    @Override
+    public List<WorkdayOpenAccountDetail> getWorkdayOpenAccountDetail() {
+        try {
+            RowMapper<WorkdayOpenAccountDetail> rowMapper = new RowMapper<WorkdayOpenAccountDetail> () {
+                public WorkdayOpenAccountDetail mapRow(ResultSet resultSet, int rowNumber) throws SQLException {
+                    WorkdayOpenAccountDetail detail = new WorkdayOpenAccountDetail();
+                    detail.setChart(resultSet.getString(FIN_COA_CD));
+                    detail.setAccountNumber(resultSet.getString(ACCOUNT_NBR));
+                    detail.setAccountName(resultSet.getString(ACCOUNT_NM));
+                    detail.setSubFundGroupWageIndicator(resultSet.getString(SUB_FUND_GRP_WAGE_IND));
+                    detail.setSubFundGroupCode(resultSet.getString(SUB_FUND_GRP_CD));
+                    detail.setHigherEdFunctionCode(resultSet.getString(FIN_HGH_ED_FUNC_CD));
+                    detail.setAccountEffectiveDate(resultSet.getDate(ACCT_EFFECT_DT));
+                    detail.setAccountClosedIndicator(resultSet.getString(ACCT_CLOSED_IND));
+                    detail.setAccountTypeCode(resultSet.getString(ACCT_TYP_CD));
+                    detail.setSubAccountNumber(resultSet.getString(SUB_ACCT_NM));
+                    detail.setSubAccountName(resultSet.getString(SUB_ACCT_NM));
+                    detail.setSubAccountActiveIndicator(resultSet.getString(SUB_ACCT_ACTV_CD));
+                    detail.setObjectCode(resultSet.getString(FIN_OBJECT_CD));
+                    detail.setSubObjectCode(resultSet.getString(FIN_SUB_OBJ_CD));
+                    detail.setSubObjectName(resultSet.getString(FIN_SUB_OBJ_CD_NM));
+                    return detail;
+                }
+            };
+            return this.getJdbcTemplate().query(buildFullOpenAccountSql(), rowMapper);
+        } catch (Exception e) {
+            LOG.error("getWorkdayOpenAccountDetail, had an error getting open account details: ", e);
+            throw new RuntimeException(e);
+        }
+    }
+    
+    protected String buildFullOpenAccountSql() {
```

Comment:
This SQL seems fairly complex, but if it's not practical to rewrite the SQL as an OJB report query (or queries), then I'm okay with leaving it in SQL form for now.

Reply:
I've not done an OJB report query before.  Since this is rather complex SQL, I would like to avoid re-writing it unless you think OJB report writer would help it be more maintainable over time.  If you think it would be more maintainable, do you know an example of using this?

Reply:
There are some simple examples in our code, such as in CuPaymentRequestDaoOjb. However, I don't think report queries allow for specifying "UNION" unless you use an OJB Criteria feature for inserting plain SQL manually.

Given that, I'm fine with leaving this query as plain SQL for now, and we can change it at a future date if necessary.